### PR TITLE
fix(l1): apply fork choice update when importing a chain

### PR DESCRIPTION
**Pull request by fmoletta**
**Motivation**

I was running the `sync` hive test suite where a chain is imported from a file and then the latest block is requested and the test failed as the latest block provided was the genesis block instead of the top block of the chain file.

**Description**

Call `apply_fork_choice` after adding the imported chain blocks in order to set the chain as canonical + update the latest block

Closes #issue_number

### CODE REVIEW

**Comment by mpaulucci** [comment]
We should mark them all as canonical. Maybe there is a helper function for this already (making a range of blocks as canonical)?

**Comment by mpaulucci** [comment]
Hmm, another issue might be that the simulation is pre merge?

File: cmd/ethereum_rust/ethereum_rust.rs [hunk 1]
```diff
@@ -118,23 +118,33 @@ async fn main() {
...
             );
-            if add_block(&block, &store).is_err() {
+            if add_block(block, &store).is_err() {
                 warn!(
                     "Failed to add block {} with hash {:#x}.",
                     block.header.number, hash
                 );
             }
         }
+        if let Some(last_block) = &blocks.last() {
```
Comment (mpaulucci) [inline cmd/ethereum_rust/ethereum_rust.rs:137]: calling apply fork choice function should do the trick.

Comment (fmoletta) [reply cmd/ethereum_rust/ethereum_rust.rs]: It works!

File: cmd/ethereum_rust/ethereum_rust.rs [hunk 2]
```diff
@@ -122,20 +123,24 @@ async fn main() {
...
+            if add_block(block, &store).is_err() {
                 warn!(
                     "Failed to add block {} with hash {:#x}.",
                     block.header.number, hash
                 );
             }
         }
         info!("Added {} blocks to blockchain", size);
+        if let Some(last_hash) = blocks.last().map(|b| b.header.compute_block_hash()) {
+            apply_fork_choice(&store, last_hash, H256::zero(), H256::zero())
```
Comment (mpaulucci) [inline cmd/ethereum_rust/ethereum_rust.rs:141]: I would probably pass `last_hash` to the 3 args instead of zero.

**Comment by fmoletta** [comment]
This change has already been included in #980 so we can close this

**Pull request closed without merging**